apps/mstpcap/main.c

```diff
@@ -159,14 +160,15 @@ static uint32_t timeval_diff_ms(struct timeval *old, struct timeval *now)
     return ms;
 }
 
-static void mstp_monitor_i_am(uint8_t mac, uint8_t *pdu, uint16_t pdu_len)
+static void mstp_monitor_i_am(
+    uint8_t mac, const uint8_t *pdu, uint16_t pdu_len)
```

Comment:
No way. That is just non sense to make everything const if you can. To me it always looks like error if I see const int without pointer mark. I have encounter many bugs by seeing const int and it should actually be const int*. Here is good read about the subject [1]. 

[1]: https://quuxplusone.github.io/blog/2022/01/23/dont-const-all-the-things/

Reply:
I like the logic of easily seen anti-pattern.

Reply:
Well in the language if course the '* is to say pointer and the const is to say 'not intended to change. Well adding it some extra meaning like 'i truly truly intend the *' I deem is rather limiting. I wonder how much advantage nowadays optimisers take of the const keyword. I am not the biggest fan of using function params as variables but would still be able to fall asleep someone using lets say pdu_len one in a loop somewhere in the function ;) Adding a const in front of it would indicate clear intention so when altering the func' to modify lets say the pdu_len before its original use in the fun' would probably spare you a bug - basically it is a way of saying hey! something in this function relies on the original value of pdu_len.

Reply:
Yeah but then you would be constantly changing function declarations. Let's say you first though you would not need change pdu_len so you made it const. Later on you notice that hey it is nice if I can change it. Now you need to change public function declaration again. That is not nice for end user of the library. They do not care if pdu_len is changed inside function or not. They just care if we change value of pointer.

If our every function is like

`XXX_XXX_XXX(const* XXX, const EEE, const EEE)`

it will be really hard to see where is even pointers. I have not seen single big library which is using const way you are proposing. If you know couple please link so I can take a look. Also compiler can optimize code better when code is const correct but when everything is const it can actually be worse.

So overall my opinion is that adding const to every function argument if we can it make reading function declaration so much harder that it would slow down reading code much more than it would save time. Developers read code like 90% of the time so it is important to able to read fast.

Reply:
I would prefer that `const` is only in front of pointers in function declarations, as @teksturi suggested.